[BPK-1316] Upgrade theo

matthewdavidson opened the pull request.

matthewdavidson commented on packages/bpk-tokens/formatters/bpk.android.xml.js:
```diff
@@ -59,8 +59,8 @@ export const tokenTemplate = ({ name, value, type, category }) =>
-export default tokens => {
-  const { props } = sortTokens(tokens);
+export default result => {
+  const { props } = sortTokens(result.toJS());
```
Custom formatters api has changed. `registerFormat` now accepts a function with an ImmutableJS Map instead of a plain ole' object. See https://github.com/salesforce-ux/theo#custom-format-function.

k0nserv commented on packages/bpk-tokens/tokens/base.raw.android.json:
```diff
@@ -309,1108 +309,848 @@
       "type": "size",
       "category": "line-heights",
-      "meta": {
-        "deprecated": true
```
This is also gone. I wouldn't think anyone relies on this metadata though, it's more for us in the source

matthewdavidson commented:
@k0nserv i've rebased this on to master (after merging https://github.com/Skyscanner/backpack/pull/650).

I've also added an entry to `unreleased.md`
